src/main/types.ts

```diff
@@ -23,10 +24,14 @@ export interface RequestConfig {
     retryDelay: number;
     statusCodesToRetry: number[][];
     headers: RequestHeaders;
-    fetch(url: string, fetchOptions: FetchOptions): Promise<Response>;
+    fetch: Fetch | FetchMock,
```

Comment:
All good, except for this one. I expect `Fetch` to be an interface, and `FetchMock` to fulfil (extend) it.

Here it should 100% remain `Fetch`, to allow for other implementations (e.g. Browser's fetch).

Reply:
My thought was that if the other implementation matches `Fetch`'s interface it can be used as it should be either `Fetch` or `FetchMock`, but in fact here `FetchMock` is "the other implementation" so I shouldn't specify it here. Good point, thanks again 🙏